query/address_search_using_ids.js

```diff
@@ -0,0 +1,175 @@
+const peliasQuery = require('pelias-query');
+const defaults = require('./search_defaults');
+const logger = require('pelias-logger').get('api');
+const _ = require('lodash');
+const check = require('check-types');
+
+//------------------------------
+// general-purpose search query
+//------------------------------
+const addressUsingIdsQuery = new peliasQuery.layout.AddressesUsingIdsQuery();
+
+// scoring boost
+addressUsingIdsQuery.score( peliasQuery.view.focus_only_function( peliasQuery.view.phrase ) );
+// --------------------------------
+
+// non-scoring hard filters
+addressUsingIdsQuery.filter( peliasQuery.view.boundary_country );
+addressUsingIdsQuery.filter( peliasQuery.view.boundary_circle );
+addressUsingIdsQuery.filter( peliasQuery.view.boundary_rect );
+addressUsingIdsQuery.filter( peliasQuery.view.sources );
+// --------------------------------
+
+
+// Red Lion, PA -- parsed as locality/state, localadmin/state, and neighbourhood/state
+// Chelsea -- parsed as neighbourhood, localadmin, and locality
+// Manhattan -- parsed as borough, locality, and localadmin
+// Luxembourg -- parsed as country, locality, and region
+
+// if any placeholder results are at neighbourhood, borough, locality, or localadmin layers, filter by those ids at those layers
+// fallback to county
+// if any placeholder results are at county or macrocounty layers, filter by those ids at those layers
+// fallback to region
+// if any placeholder results are at region or macroregion layers, filter by those ids at those layers
+// fallback to dependency/country
+// if any placeholder results are at dependency or country layers, filter by those ids at those layers
+
+
+// address in Red Lion, PA -- find results at layer=address
+// neighbourhood_id in [85844063, 85844067]
+// locality_id in [101717221]
+// localadmin_id in [404487867]
+// search all of the above
+
+// address in Chelsea
+// neighbourhood_id in [85786511, 85810589, 85769021, 85890029, 85810579, 85810591, 85810575, 85772883, 420514219]
+// locality_id in [85950359, 85914491, 101932747, 85951865, 101715289, 85943049, 101733697, 101722101, 101738587]
+// localadmin_id in [404476575, 404508239, 404474971, 404527169, 404494675, 404503811, 404519887, 404488679, 404538119]
+
+// address in Manhattan
+// neighbourhood_id in []
+// borough_id in [421205771]
+// locality_id in [85945171, 85940551, 85972655]
+// localadmin_id in [404502889, 404499147, 404502891, 85972655]
+// search all of the above
+
+// address in Luxembourg
+// country_id in [85633275]
+// region_id in [85681727, 85673875]
+// locality_id in [101751765]
+// search locality first, then region perhaps
+
```

Comment:
these dev notes are great, can we clean them up a bit to make them easier to understand for a first-time contributor viewing this for the first time?